blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -0,0 +1,192 @@
+package org.http4s
+package client
+package blaze
+
+import cats.effect._
+import cats.implicits._
+import fs2.Stream
+import java.nio.channels.AsynchronousChannelGroup
+import javax.net.ssl.SSLContext
+import org.http4s.headers.{AgentProduct, `User-Agent`}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+sealed abstract class BlazeClientBuilder[F[_]] private (
+    val responseHeaderTimeout: Duration,
+    val idleTimeout: Duration,
+    val requestTimeout: Duration,
+    val userAgent: Option[`User-Agent`],
+    val maxTotalConnections: Int,
+    val maxWaitQueueLimit: Int,
+    val maxConnectionsPerRequestKey: RequestKey => Int,
+    val sslContext: Option[SSLContext],
+    val checkEndpointIdentification: Boolean,
+    val maxResponseLineSize: Int,
+    val maxHeaderLength: Int,
+    val maxChunkSize: Int,
+    val parserMode: ParserMode,
+    val bufferSize: Int,
+    val executionContext: ExecutionContext,
```

Comment:
I'd like to just use a `ContextShift` here, but blaze needs an `ExecutionContext`.  This is kind of leaking its `Future` implementation in a way.

Reply:
Either we do this, or we rely on some config somewhere.